crates/wysiwyg/src/composer_model/delete_text.rs

```diff
@@ -236,6 +236,8 @@ where
                     _ => ComposerUpdate::keep(),
                 }
             }
+            DomNode::Mention(_) => self
+                .delete_to_cursor(direction.increment(location.index_in_dom())),
```

Comment:
I totally agree - I actually unpicked those changes ([see revert](https://github.com/matrix-org/matrix-rich-text-editor/pull/699/commits/e34644605c2686ffd620c6b453b7e87394d367d5)) so that we can maintain compatibility for non editable links. But we can definitely clean this up when we drop support for that.